apis/apps/v1alpha1/cluster_types.go

```diff
@@ -612,6 +604,14 @@ type ClusterComponentSpec struct {
 	// The InstanceTemplate provides a way to override values in the default template,
 	// allowing the component to manage instances from different templates.
 	//
+	// The naming convention for instances (pods) based on the Component Name, InstanceTemplate Name, and ordinal.
+	// The constructed instance name follows the pattern: $(component.name)-$(template.name)-$(ordinal).
+	// By default, the ordinal starts from 0 for each InstanceTemplate.
+	// It is important to ensure that the Name of each InstanceTemplate is unique.
+	//
+	// The sum of replicas across all InstanceTemplates should not exceed the total number of Replicas specified for the Component.
+	// Any remaining replicas will be generated using the default template and will follow the default naming rules.
+	//
 	// +optional
 	Instances []InstanceTemplate `json:"instances,omitempty"`
```

Comment:
you can you add  add  `patchStrategy:"merge,retainKeys" patchMergeKey:"name"` to verify that `template name` is the unique key

Reply:
fixed in: [1d29e88](https://github.com/apecloud/kubeblocks/pull/6958/commits/1d29e8829b0f2c0709a5ab0492f0766082f6500c)